ext/ffi/00_ffi.js

```diff
@@ -120,141 +115,28 @@
-          );
-        }
-        ArrayPrototypePush(parameters, arg);
-      } else if (type === "u64" || type === "usize") {
-        if (
-          !(NumberIsInteger(arg) && arg >= 0 ||
-            typeof arg === "bigint" && 0n <= arg && arg <= 0xffffffffffffffffn)
-        ) {
-          throw new TypeError(
-            `Expected FFI argument to be an unsigned integer, but got '${arg}'`,
-          );
-        }
-        ArrayPrototypePush(parameters, arg);
-      } else if (type == "i64" || type === "isize") {
-        if (
-          !(NumberIsInteger(arg) ||
-            typeof arg === "bigint" && -1n * 2n ** 63n <= arg &&
-              arg <= 2n ** 63n - 1n)
-        ) {
-          throw new TypeError(
-            `Expected FFI argument to be a signed integer, but got '${arg}'`,
-          );
-        }
-        ArrayPrototypePush(parameters, arg);
-      } else if (type === "f32" || type === "f64") {
-        if (!NumberIsFinite(arg)) {
-          throw new TypeError(
-            `Expected FFI argument to be a number, but got '${arg}'`,
-          );
-        }
-        ArrayPrototypePush(parameters, arg);
-      } else if (type === "pointer") {
-        if (
-          ObjectPrototypeIsPrototypeOf(ArrayBufferPrototype, arg?.buffer) &&
-          arg.byteLength !== undefined
-        ) {
-          ArrayPrototypePush(parameters, arg);
-        } else if (ObjectPrototypeIsPrototypeOf(UnsafePointerPrototype, arg)) {
-          ArrayPrototypePush(parameters, arg.value);
-        } else if (arg === null) {
-          ArrayPrototypePush(parameters, null);
-        } else {
-          throw new TypeError(
-            "Expected FFI argument to be TypedArray, UnsafePointer or null",
-          );
-        }
-      } else if (
-        typeof type === "object" && type !== null && "function" in type
-      ) {
-        if (ObjectPrototypeIsPrototypeOf(UnsafeCallbackPrototype, arg)) {
-          // Own registered callback, pass the pointer value
-          ArrayPrototypePush(parameters, arg.pointer.value);
-        } else if (arg === null) {
-          // nullptr
-          ArrayPrototypePush(parameters, null);
-        } else if (
-          ObjectPrototypeIsPrototypeOf(UnsafeFnPointerPrototype, arg)
-        ) {
-          // Foreign function, pass the pointer value
-          ArrayPrototypePush(parameters, arg.pointer.value);
-        } else if (
-          ObjectPrototypeIsPrototypeOf(UnsafePointerPrototype, arg)
-        ) {
-          // Foreign function, pass the pointer value
-          ArrayPrototypePush(parameters, arg.value);
-        } else {
-          throw new TypeError(
-            "Expected FFI argument to be UnsafeCallback, UnsafeFnPointer, UnsafePointer or null",
-          );
-        }
-      } else {
-        throw new TypeError(`Invalid FFI argument type '${type}'`);
+    static of(value) {
+      if (ObjectPrototypeIsPrototypeOf(UnsafeCallbackPrototype, value)) {
```

Comment:
Supporting UnsafeCallbacks for `UnsafePointer.of` seems unnecessary.

Reply:
We will get rid of the UnsafePointer class completely. I was thinking that we could turn this into a more common interface until it's removed.